heron/downloaders/src/java/com/twitter/heron/downloader/FileDownloader.java

```diff
@@ -0,0 +1,26 @@
+//  Copyright 2017 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package com.twitter.heron.downloader;
+
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+
+public class FileDownloader implements Downloader {
+  @Override
+  public void download(URI uri, Path destination) throws Exception {
+    final URL url = uri.toURL();
```

Comment:
do we need try/cache here?

Reply:
though this is really done once on the start of the topology

Reply:
kk.